packages/runtime/src/utils/share.ts

```diff
@@ -171,59 +172,54 @@ export function getRegisteredShare(
       localShareScopeMap[sc][pkgName]
     ) {
       const { requiredVersion } = shareConfig;
-      // eslint-disable-next-line max-depth
-      if (shareConfig.singleton) {
-        const singletonVersion = getFindShareFunction(strategy)(
-          localShareScopeMap,
-          sc,
-          pkgName,
-        );
-        if (typeof singletonVersion === 'function') {
-          return singletonVersion({ localShareScopeMap, sc, pkgName });
-        }
-        // eslint-disable-next-line max-depth
-        if (
-          typeof requiredVersion === 'string' &&
-          !satisfy(singletonVersion, requiredVersion)
-        ) {
-          warn(
-            `Version ${singletonVersion} from ${
-              singletonVersion &&
-              localShareScopeMap[sc][pkgName][singletonVersion].from
-            } of shared singleton module ${pkgName} does not satisfy the requirement of ${
-              shareInfo.from
-            } which needs ${requiredVersion})`,
-          );
-        }
-
-        return localShareScopeMap[sc][pkgName][singletonVersion];
-      } else {
-        const maxVersion = getFindShareFunction(strategy)(
-          localShareScopeMap,
-          sc,
-          pkgName,
-        );
-
-        // eslint-disable-next-line max-depth
-        if (requiredVersion === false || requiredVersion === '*') {
-          return localShareScopeMap[sc][pkgName][maxVersion];
-        }
-
-        // eslint-disable-next-line max-depth
-        if (satisfy(maxVersion, requiredVersion)) {
-          return localShareScopeMap[sc][pkgName][maxVersion];
-        }
+      const findShareFunction = getFindShareFunction(strategy);
+      const maxOrSingletonVersion = findShareFunction(
+        localShareScopeMap,
+        sc,
+        pkgName,
+      );
+
+      //@ts-ignore
+      const defaultResolver = () => {
```

Comment:
not all code paths return, unsure if refactor is problem or if i should use original function implementation, just wrap in defaultResolver factory?